.travis.yml

```diff
@@ -1,27 +1,50 @@
 language: c
 env:
-  - OCAML_VERSION=4.09.0
+  - TO_TEST=changes
+  - TO_TEST=install OCAML_VERSION=4.07.1
+  - TO_TEST=install OCAML_VERSION=4.08.0
+  - TO_TEST=tests OCAML_VERSION=4.09.0
 cache:
   directories:
     - ${HOME}/.opam
 before_install:
-  - wget -O ${HOME}/opam https://github.com/ocaml/opam/releases/download/2.0.2/opam-2.0.2-x86_64-linux
-  - chmod +x ${HOME}/opam
-  - export PATH=${HOME}:${PATH}
-  - export OPAMYES=1
-  - export OPAMJOBS=2
-  - opam init --bare --disable-sandboxing
-  - opam switch --dry-run ${OCAML_VERSION} 2>/dev/null || opam switch create ${OCAML_VERSION}
+  - |
+    case "$TO_TEST" in
+      install|tests)
+        wget -O ${HOME}/opam https://github.com/ocaml/opam/releases/download/2.0.2/opam-2.0.2-x86_64-linux
+        chmod +x ${HOME}/opam
+        export PATH=${HOME}:${PATH}
+        export OPAMYES=1
+        export OPAMJOBS=2
+        opam init --bare --disable-sandboxing
+        opam switch --dry-run ${OCAML_VERSION} 2>/dev/null || opam switch create ${OCAML_VERSION}
+        ;;
+    esac
 install:
-  - opam switch ${OCAML_VERSION}
-  - opam update --upgrade
-  - opam pin add --no-action ocamlformat .
-  - opam install --deps-only ocamlformat
+  - |
+    case "$TO_TEST" in
+      install|tests)
+        opam switch ${OCAML_VERSION}
+        opam update --upgrade
+        opam pin add --no-action ocamlformat .
```

Comment:
We should do that below, in `script`. Otherwise it may be cached.

Reply:
done